### ISSUE TITLE
CBOR serialization planned?

### ISSUE THREAD

**Issue by EyeCon** [korlibs/korio#67]
Hello,

Are there any plans to support CBOR serialization? It is a binary-only, compact and well-designed serialization format.

References:

* https://cbor.io/
* https://en.wikipedia.org/wiki/CBOR
* https://tools.ietf.org/html/rfc7049

Thanks in advance.

**Comment by soywiz** [comment]
Was not planned. But we can add it at some point. If someone wants to make a PR, this will happen sooner

**Comment by RaphaelTarita** [comment]
AFAIK there's now a CBOR format backend for kotlinx-serialization: https://www.github.com/Kotlin/kotlinx.serialization/tree/master/formats%2FREADME.md

Which would make another implementation superfluous, or am I not getting the point here?

**Comment by soywiz** [comment]
I don't believe it is superfluous. In fact, CBOR is schema-less. Korio also supports easy to use dynamic access to unstructured data in common targets which is a different use case.
And not checked the latest version, but kotlinx.serialization used to add a lot of extra size to the output by just adding a few serializable classes. Not that much as adding kotlin-reflect, but enough to consider it on simpler use cases.

This is not on my priorities right now, but I think it is totally reasonable to do that, so if anyone wants to contribute with this I will accept PRs 👍
